packages/babel-generator/src/source-map.ts

```diff
@@ -76,24 +102,56 @@ export default class SourceMap {
     line: number,
     column: number,
     identifierName?: string | null,
+    identifierNamePos?: { line: number; column: number },
     filename?: string | null,
   ) {
     this._rawMappings = undefined;
 
+    let original;
+    let source;
+
+    if (line != null) {
+      if (this._inputMap) {
+        if (identifierNamePos) {
+          const name = originalPositionFor(this._inputMap, {
+            line,
+            column,
+          }).name;
+          if (name) {
+            identifierName = name;
```

Comment:
Nit: If we're prioritizing the below lookup's name, we can skip this if that one found a name.

Reply:
But the original position of the parentheses will be pointed to the identifier, what do you think of this?

Reply:
I'm not suggesting we take the original `line`/`column` from this lookup, but that we skip this lookup entirely.

```js
// This is the lookup for this mark
const originalMapping = originalPositionFor(this._inputMap, {
  line,
  column,
});

if (originalMapping.name) {
  // If the we found a name, nothing else needs to be done
  // Maybe we're marking a `(` and the input map already had a name attached there,
  // or we're marking a `(` and the sourcemap spanned a `foo(`,
  // or we're marking an identifier, etc.
  identifierName = originalMapping.name;
} else if (identifierNamePos) {
  // We're trying to mark a `(` (as that's the only thing that provides
  // an identifierNamePos currently), and we the AST had an identifier attached.
  // Lookup it's original name.
  const originalIdentifierMapping = originalPositionFor(this._inputMap, identifierNamePos);
  if (originalIdentifierMapping.name) {
    identifierName = originalIdentifierMapping.name;
  }
} else {
  // noop, just use the passed in identifierName
}

Reply:
It's all really great! I also left these amazing comments in the code.